app-codepipeline/codepipeline.tf

```diff
@@ -2,18 +2,18 @@
 data "aws_caller_identity" "current" {}
 locals {
   ecr_account_id = var.ecr_account_id == "" ? data.aws_caller_identity.current.account_id : var.ecr_account_id
-  ecr_namespace  = var.ecr_namespace == "" ? "${var.name_prefix}-${var.repo.branch}" : var.ecr_namespace
-  # build_env_vars = list(object({"name"=string, "value"=string}))
+  ecr_namespace  = var.ecr_namespace == "" ? "${var.name_prefix}-${var.sdlc_stage}-${var.repo.branch}" : var.ecr_namespace
   build_env_vars = concat([
     { "name" : "ECR_REGION", "value" : "${var.aws_region}" },
     { "name" : "ECR_ACCOUNT_ID", "value" : "${local.ecr_account_id}" },
     { "name" : "ECR_NAMESPACE", "value" : "${local.ecr_namespace}" }
   ], [for n, val in var.build_env_vars : { name = n, value = val }])
+  name_prefix = "${var.name_prefix}-${var.sdlc_stage}"
```

Comment:
Again, this pipeline spans multiple stages/environments.  Why are we putting a single value into the name prefix used by ostensibly all resources, including staging and prod?

Reply:
I see- I was under the impression that we were planning on invoking codepipeline per sdlc_stage. I see how we'd add staging and prod, using new codepipeline stage resources instead. I'll close this